src/core/device.py

```diff
@@ -78,16 +89,38 @@ def connect(self):
                 # port.close()
                 # print("We are reading something wrong. Are you trying to communicate with another device?", file=sys.stderr)
                 # exit(-1)
-            self.reader = SchwinnReader(self.port, self.backup)
+            if sign1 == 0x243601:  #kb: think maybe we need something else to identify soleus.  This might be specific to my watch.  Perhaps the M11165 string?
```

Comment:
But all that kind of info is usually available way after initial connect. Cresta used way shorter chunks of data vs Schwinn as it had no altitude IIRC. We need to document formats. I have it sitting somewhere locally for Schwinn :-)